libakumuli/storage_engine/blockstore.h

```diff
@@ -117,37 +117,72 @@ class FixedSizeFileStorage : public BlockStore,
     std::vector<std::string> volume_names_;
 
     //! Secret c-tor.
-    FixedSizeFileStorage(std::string metapath, std::vector<std::string> volpaths);
+    FileStorage(std::string metapath, std::vector<std::string> volpaths);
 
+    virtual void update_current_volume() = 0;
```

Comment:
The naming isn't great, agreed. I ended up factoring out the function as the only part of advance_volume() that differs is how one updates the info about current volume. For fixed it's just about updating the current_volume_ using modulo on num vols. For expandable it's incrementing it and creating the new volume and updating the meta data book keeping. I'll think of a better name, please do the same!